packages/zoe/src/contracts/callSpread.js

```diff
@@ -0,0 +1,217 @@
+      const totalCollateral = terms.settlementAmount;
+      const collateralShare = floorDivide(
+        multiply(totalCollateral.value, sharePercent),
+        PERCENT_BASE,
+      );
+      const seatPortion = collateralMath.make(collateralShare);
+      const collateralRemainder = collateralMath.subtract(
+        currentCollateral,
+        seatPortion,
+      );
+      zcf.reallocate(
+        seat.stage({ Collateral: seatPortion }),
+        collateralSeat.stage({ Collateral: collateralRemainder }),
+      );
+      seat.exit();
+    });
+  }
+
+  // calculate the portion (as a percentage) of the collateral that should be
+  // allocated to the long side.
+  function calculateLongShare(price) {
+    if (strikeMath.isGTE(terms.strikePrice1, price)) {
+      return 0;
+    } else if (strikeMath.isGTE(price, terms.strikePrice2)) {
+      return PERCENT_BASE;
+    }
+
+    const denominator = strikeMath.subtract(
+      terms.strikePrice2,
+      terms.strikePrice1,
+    ).value;
+    const numerator = strikeMath.subtract(price, terms.strikePrice1).value;
+    return floorDivide(multiply(PERCENT_BASE, numerator), denominator);
+  }
+
+  function payoffOptions(price) {
+    // either offer might be exercised late, so we pay the two seats separately.
+    const longShare = calculateLongShare(price);
+    reallocateToSeat(Position.LONG, longShare);
+    reallocateToSeat(Position.SHORT, inverse(longShare));
+  }
+
+  function schedulePayoffs() {
+    terms.priceAuthority
+      .priceAtTime(terms.expiration, terms.underlyingAmount)
+      .then(price => payoffOptions(price));
+  }
+
+  function makeOptionInvitation(dir) {
+    const optionsTerms = harden({
+      ...terms,
+      position: dir,
+    });
+    // All we do at time of exercise is resolve the promise.
+    return zcf.makeInvitation(
+      seat => seatPromiseKits[dir].resolve(seat),
+      `collect ${dir} payout`,
+      optionsTerms,
+    );
+  }
+
+  async function makeOptionPair() {
+    return {
+      longInvitation: makeOptionInvitation(Position.LONG),
+      shortInvitation: makeOptionInvitation(Position.SHORT),
+    };
+  }
+
+  async function makeInvitationToBuy() {
+    const { longInvitation, shortInvitation } = await makeOptionPair();
+    const invitationIssuer = zcf.getInvitationIssuer();
+    const longAmount = await E(invitationIssuer).getAmountOf(longInvitation);
+    const shortAmount = await E(invitationIssuer).getAmountOf(shortInvitation);
+    depositToSeat(
+      zcf,
+      collateralSeat,
+      { LongOption: longAmount, ShortOption: shortAmount },
+      { LongOption: longInvitation, ShortOption: shortInvitation },
+    );
+
+    // transfer collateral from longSeat to collateralSeat, then return a pair
+    // of callSpread invitations
+    /** @type {OfferHandler} */
+    const pairBuyerPosition = longSeat => {
+      assertProposalShape(longSeat, {
+        give: { Collateral: null },
+        want: { LongOption: null, ShortOption: null },
+      });
+
```

Comment:
For simplicity's sake, can we remove the two option invitations from the allocation for the creatorSeat? What I'm envisioning is similar to how `atomicSwap` returns the second participant's invitation - through the first invitation's `offerResult`.

Reply:
I agree that that would make a simpler contract, but it feels like a false simplicity to me. I don't think we would recommend that anyone use the version that doesn't provide offer safety to the creator.

I don't think we need more trivial examples, we need more simple examples that do just enough.

Do you think it would be a simpler example if the contract held onto the two invitations as local state rather than using `depositToSeat()`? It could then `depositToSeat()` directly to the seat in `reallocateToSeat()`.

I think offerSafety is handled better with the invitations in the collateralSeat's allocation, and a call to `reallocate()` moves them to the seat.

Reply:
> I agree that that would make a simpler contract, but it feels like a false simplicity to me. I don't think we would recommend that anyone use the version that doesn't provide offer safety to the creator.

That makes a lot of sense to me.